components/support/migration/src/main/java/mozilla/components/support/migration/FennecSessionMigration.kt

```diff
@@ -73,3 +74,48 @@ internal object FennecSessionMigration {
         return files
     }
 }
+
+private fun Result.Success<SessionManager.Snapshot>.filter(
+    logger: Logger
+): Result.Success<SessionManager.Snapshot> {
+    var selectedIndex = value.selectedSessionIndex
+
+    logger.debug("Filtering migrated sessions (size=${value.sessions.size}, index=$selectedIndex)")
+
+    val sessions = value.sessions.mapIndexedNotNull { index, item ->
+        when {
+            // We filter out about:home tabs since Fenix does not handle those URLs.
+            item.session.url == ABOUT_HOME -> {
+                logger.debug("- Filtering about:home URL")
+                if (index < selectedIndex) {
+                    selectedIndex--
+                }
+                null
+            }
+
+            // We rewrite about:reader URLs to their actual URLs. Fenix does not handle about:reader
+            // URLs, but it can load the original URL.
+            item.session.url.startsWith(ABOUT_READER) -> {
+                val url = Uri.decode(item.session.url.substring(ABOUT_READER.length))
```

Comment:
Any chance Uri.decode can fail? If so, we'll fail the whole session store parse, vs just ignoring an individual (bad) url.

Reply:
Just read through the code and saw that it could theoretically at least throw `IllegalStateException`. I think I'll just catch `Exception` here and report this to Sentry.